Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp

```diff
@@ -2493,3 +2494,14 @@ bool CLuaPedDefs::SetPedExitVehicle(CClientPed* pPed)
 {
     return pPed->ExitVehicle();
 }
+
+bool CLuaPedDefs::killPedTask(CClientPed* ped, std::string_view taskType, std::uint8_t taskNumber, std::optional<bool> gracefully) noexcept
```

Comment:
It looks like you need enum instead of string here

Reply:
but getPedTask ( ped theMed, string priority, int taskType ) use string so should we have similar structure?

Reply:
It will be a string in Lua API, but we are using enums for such data

Reply:
I still do not understand it can u give me the code or something explain more

Reply:
check `CLuaEngineDefs::EngineSetPoolCapacity`

Reply:
I am getting this error with the argument parser
   F:\MTA\mtasa-blue\Shared\mods\deathmatch\logic\lua\CLuaFunctionParser.h(426,54):
4>            Conversion to enumeration type requires an explicit cast (static_cast, C-style cast or parenthesized function-style cast)
4>    F:\MTA\mtasa-blue\Shared\mods\deathmatch\logic\lua\CLuaFunctionParser.h(426,35):
4>    while trying to match the argument list '(T)'
4>        with
4>        [
4>            T=eSecondaryTaskType
4>        ]

Reply:
I have fix it and add the enum @TheNormalnij